### ISSUE TITLE
Error with Style/MapToHash that's locally disabled

### ISSUE THREAD

**Issue by ccutrer** [rubocop/rubocop#12558]
## Expected behavior

Not to get an error.

## Actual behavior

An error occurred, and also an additional offense is register for an unnecessary disabling.

## Steps to reproduce the problem

`.rubocop.yml`:
```yaml
AllCops:
  NewCops: enable
```

`test.rb`:
```ruby
# frozen_string_literal: true

module Java::JavaUtil::Map # rubocop:disable Style/ClassAndModuleChildren
  def transform_values
    map do |k, v| # rubocop:disable Style/MapToHash, Style/HashTransformValues
      [k, (yield v)]
    end.to_h
  end
end
```

```
$ rubocop -d test.rb

For /Users/cody/src: configuration from /Users/cody/src/.rubocop.yml
Default configuration from /Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.59.0/config/default.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning /Users/cody/src/test.rb
An error occurred while Style/MapToHash cop was inspecting /Users/cody/src/test.rb:5:4.
undefined method `source' for nil:NilClass

          message = format(MSG, method: map_node.loc.selector.source, dot: map_node.loc.dot.source)
                                                                                           ^^^^^^^
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.59.0/lib/rubocop/cop/style/map_to_hash.rb:55:in `on_send'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.59.0/lib/rubocop/cop/commissioner.rb:143:in `public_send'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.59.0/lib/rubocop/cop/commissioner.rb:143:in `block (2 levels) in trigger_restricted_cops'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.59.0/lib/rubocop/cop/commissioner.rb:171:in `with_cop_error_handling'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.59.0/lib/rubocop/cop/commissioner.rb:142:in `block in trigger_restricted_cops'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.59.0/lib/rubocop/cop/commissioner.rb:141:in `each'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.59.0/lib/rubocop/cop/commissioner.rb:141:in `trigger_restricted_cops'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.59.0/lib/rubocop/cop/commissioner.rb:70:in `on_send'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-ast-1.30.0/lib/rubocop/ast/traversal.rb:154:in `on_def'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.59.0/lib/rubocop/cop/commissioner.rb:71:in `on_def'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-ast-1.30.0/lib/rubocop/ast/traversal.rb:138:in `on_while'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.59.0/lib/rubocop/cop/commissioner.rb:71:in `on_module'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-ast-1.30.0/lib/rubocop/ast/traversal.rb:20:in `walk'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.59.0/lib/rubocop/cop/commissioner.rb:87:in `investigate'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.59.0/lib/rubocop/cop/team.rb:156:in `investigate_partial'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.59.0/lib/rubocop/cop/team.rb:98:in `investigate'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.59.0/lib/rubocop/runner.rb:345:in `block in inspect_file'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.59.0/lib/rubocop/runner.rb:344:in `each'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.59.0/lib/rubocop/runner.rb:344:in `flat_map'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.59.0/lib/rubocop/runner.rb:344:in `inspect_file'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.59.0/lib/rubocop/runner.rb:287:in `block in do_inspection_loop'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.59.0/lib/rubocop/runner.rb:321:in `block in iterate_until_no_changes'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.59.0/lib/rubocop/runner.rb:314:in `loop'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.59.0/lib/rubocop/runner.rb:314:in `iterate_until_no_changes'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.59.0/lib/rubocop/runner.rb:283:in `do_inspection_loop'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.59.0/lib/rubocop/runner.rb:164:in `block in file_offenses'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.59.0/lib/rubocop/runner.rb:189:in `file_offense_cache'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.59.0/lib/rubocop/runner.rb:163:in `file_offenses'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.59.0/lib/rubocop/runner.rb:154:in `process_file'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.59.0/lib/rubocop/runner.rb:135:in `block in each_inspected_file'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.59.0/lib/rubocop/runner.rb:134:in `each'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.59.0/lib/rubocop/runner.rb:134:in `reduce'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.59.0/lib/rubocop/runner.rb:134:in `each_inspected_file'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.59.0/lib/rubocop/runner.rb:120:in `inspect_files'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.59.0/lib/rubocop/runner.rb:73:in `run'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.59.0/lib/rubocop/cli/command/execute_runner.rb:26:in `block in execute_runner'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.59.0/lib/rubocop/cli/command/execute_runner.rb:52:in `with_redirect'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.59.0/lib/rubocop/cli/command/execute_runner.rb:25:in `execute_runner'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.59.0/lib/rubocop/cli/command/execute_runner.rb:17:in `run'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.59.0/lib/rubocop/cli/command.rb:11:in `run'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.59.0/lib/rubocop/cli/environment.rb:18:in `run'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.59.0/lib/rubocop/cli.rb:118:in `run_command'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.59.0/lib/rubocop/cli.rb:125:in `execute_runners'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.59.0/lib/rubocop/cli.rb:51:in `block in run'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.59.0/lib/rubocop/cli.rb:77:in `profile_if_needed'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.59.0/lib/rubocop/cli.rb:43:in `run'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.59.0/exe/rubocop:19:in `block in <top (required)>'
/Users/cody/.rubies/ruby-3.1.3/lib/ruby/3.1.0/benchmark.rb:311:in `realtime'
/Users/cody/.gem/ruby/3.1.3/gems/rubocop-1.59.0/exe/rubocop:19:in `<top (required)>'
/Users/cody/.gem/ruby/3.1.3/bin/rubocop:25:in `load'
/Users/cody/.gem/ruby/3.1.3/bin/rubocop:25:in `<main>'
W

Offenses:

test.rb:3:1: C: Style/Documentation: Missing top-level documentation comment for module Java::JavaUtil::Map.
module Java::JavaUtil::Map # rubocop:disable Style/ClassAndModuleChildren
^^^^^^^^^^^^^^^^^^^^^^^^^^
test.rb:5:37: W: [Correctable] Lint/RedundantCopDisableDirective: Unnecessary disabling of Style/MapToHash.
    map do |k, v| # rubocop:disable Style/MapToHash, Style/HashTransformValues
                                    ^^^^^^^^^^^^^^^

1 file inspected, 2 offenses detected, 1 offense autocorrectable

1 error occurred:
An error occurred while Style/MapToHash cop was inspecting /Users/cody/src/test.rb:5:4.
Errors are usually caused by RuboCop bugs.
Please, report your problems to RuboCop's issue tracker.
https://github.com/rubocop/rubocop/issues

Mention the following information in the issue report:
1.59.0 (using Parser 3.2.2.4, rubocop-ast 1.30.0, running on ruby 3.1.3) [arm64-darwin21]
Finished in 0.11219099990557879 seconds
```

## RuboCop version

```
$ rubocop -V
1.59.0 (using Parser 3.2.2.4, rubocop-ast 1.30.0, running on ruby 3.1.3) [arm64-darwin21]
```
